far/vmenu.cpp

```diff
@@ -140,15 +140,48 @@ void VMenu::ResetCursor()
 	GetCursorType(PrevCursorVisible,PrevCursorSize);
 }
 
+template <std::ranges::random_access_range coll, std::indirect_unary_predicate<std::ranges::iterator_t<coll>> pred>
```

Comment:
`contiguous_range` maybe, since we're assuming this by taking addresses below?

`std::indirect_unary_predicate<std::ranges::iterator_t<coll>>` is truly monstrous :) Also, Sonar [doesn't like it already](https://sonarcloud.io/project/issues?resolved=false&types=BUG&pullRequest=751&id=FarGroup_FarManager&open=AYw80UcbIuJFFgMMy3f2). Maybe leave unrestricted?

P.S. Terse syntax (`func(auto concept Name)`) is also fine.

Reply:
> P.S. Terse syntax (`func(auto concept Name)`) is also fine.

Looks interesting. Could you please elaborate?

Reply:
Oh, I see. Sorry, was too distracted with thinking about the "predicate" thingy.

Reply:
I mean it can be spelled as just 
```C++
int find_nearest(std::ranges::contiguous_range auto const& Range, const int Pos, const auto Pred, const bool GoBackward, const bool DoWrap)
```
without template<> etc.

Do we still need `&&`?